src/main/webapp/script.js

```diff
@@ -5,11 +5,16 @@ function loadListings() {
   getListings();
 }
 
+//   Adjusts links on the nav bar based on whether the user is logged in
 function fetch_auth_info(){
-  fetch('/AuthServlet').then(response => response.json()).then((logUrl) => {
-  console.log(logUrl);
-  document.getElementById("loglink").href = logUrl[1];
-  document.getElementById("loglink").innerText = logUrl[0];
+    fetch('/AuthServlet').then(response => response.json()).then((logUrl) => {
+    console.log(logUrl);
+    document.getElementById("loglink").innerText = logUrl[0];
+    document.getElementById("loglink").href = logUrl[1];
+    if (logUrl[0] == "Login") {
```

Comment:
Another way I thought of restricting navigation is:

- Allow the user to navigate to any page, but always call `fetch_auth_info()` during `onload`
- For every call to `fetch_auth_info()`, include in the HttpRequest a parameter of the current page
- In the AuthServlet, if user is not logged in, then create a loginUrl with the parameter of the current page that was given, and send it to the client
- `fetch_auth_info()` then will check if user is logged in or not
  - If they are logged in, do normal navigation
  - If they are not logged in, redirect navigation to the newly created loginUrl with the parameter of the page we passed

This might create a more simple scheme that works with any page without having to manually change every link in the navbar to a new loginUrl link.

Either way works, though, and I'm fine with both. Was just offering another perspective.

Reply:
I can also hop on a Google Meet at almost any time today if u would like for me to elaborate / explain more

Reply:
Thanks Angello! I tried doing initially but I was running into bugs so I decided to do it this way. I might go back and see if I can change it though I just wanted to get something working for the MVP.